### PR TITLE
Add "keyed" behavior to some aggregations

### PR DESCRIPTION
**Pull request by romainneutron**

### CODE REVIEW

File: src/Aggregation/Range.php [hunk 1]
```diff
@@ -45,16 +47,6 @@ public function addRange($fromValue = null, $toValue = null, ?string $key = null
         return $this->addParam('ranges', $range);
     }
 
-    /**
-     * If set to true, a unique string key will be associated with each bucket, and ranges will be returned as an associative array.
-     *
-     * @return $this
-     */
-    public function setKeyed(bool $keyed): self
```
Comment (deguif) [inline src/Aggregation/Range.php:53]: Little BC break here too.

File: src/Aggregation/PercentilesBucket.php [hunk 2]
```diff
@@ -69,12 +71,4 @@ public function setPercents(array $percents): self
     {
         return $this->setParam('percents', $percents);
     }
-
-    /**
-     * Set keyed flag to return the range as an hash instead of an array of key-value pairs.
-     */
-    public function setKeyed(bool $keyed): self
```
Comment (deguif) [inline src/Aggregation/PercentilesBucket.php:76]: Little BC break here, but not so important. So can probably be ignored. ping @ruflin @thePanz for confirmation?

**Review by deguif — APPROVED** [review]
Big 👍 for using trait.

File: src/Aggregation/PercentilesBucket.php (hunk 2 above)
Comment (deguif) [reply src/Aggregation/PercentilesBucket.php:76]: Maybe just adding an entry to the BC breaks section in the changelog is fine.

Comment (romainneutron) [reply src/Aggregation/PercentilesBucket.php:76]: How can it break BC?

Comment (deguif) [reply src/Aggregation/PercentilesBucket.php:76]: Previous method signature was different than the new one (facultative argument now in the trait).

This breaks when overriding method in a subclass.
See https://3v4l.org/FPiOB for exemple.

Comment (romainneutron) [reply src/Aggregation/PercentilesBucket.php:76]: Ah okay, you're right.

File: src/Aggregation/Range.php (hunk 1 above)
Comment (romainneutron) [reply src/Aggregation/Range.php:53]: fixed

File: src/Aggregation/PercentilesBucket.php (hunk 2 above)
Comment (romainneutron) [reply src/Aggregation/PercentilesBucket.php:76]: fixed

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased](https://github.com/ruflin/Elastica/compare/7.0.0...master)
 ### Backward Compatibility Breaks
 * Allow the Terms query to accept arrays of strings, ints and floats [#1872](https://github.com/ruflin/Elastica/pull/1872)
+* Added a default value to `Elastica\Aggregation\Range::setKeyed()` and `Elastica\PercentilesBucket\Range::setKeyed()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
```
Comment (deguif) [inline CHANGELOG.md:10]:
```suggestion
* Added a default value to `Elastica\Aggregation\Range::setKeyed()` and `Elastica\Aggregation\PercentilesBucket::setKeyed()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
```

**Review by deguif — APPROVED** [review]
LGTM, but I'm waiting another review from @ruflin or @thePanz before merging it to confirm we can live with these BC breaks.

**Comment by romainneutron** [comment]
PR updated comments addressed

File: CHANGELOG.md [hunk 4]
```diff
@@ -23,6 +24,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Added `static_lambda` CS rule [#1870](https://github.com/ruflin/Elastica/pull/1870)
 * Added `Elastica\Aggregation\DateRange::setTimezone()` [#1847](https://github.com/ruflin/Elastica/pull/1847)
 * Added endpoint options support to `Elastica\Index::create()` [#1859](https://github.com/ruflin/Elastica/pull/1859)
+* Added `Elastica\Aggregation\DateHistogram::setKeyed()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
```
Comment (ruflin) [inline CHANGELOG.md:27]: A few duplicates here.

**Review by ruflin — COMMENTED** [review]
LGTM except the changelog. Seems we also got rid of the breaking changes?

**Comment by deguif** [comment]
@ruflin BC break is still there, see https://github.com/ruflin/Elastica/pull/1876#discussion_r528885157
It was added to BC break section in changelog.

File: CHANGELOG.md (hunk 4 above)
Comment (ruflin) [reply CHANGELOG.md:27]: Ignore me, same issue as on the other PR, me being blind.

**Review by ruflin — APPROVED** [review]
I'm good with this BC change.

**Comment by deguif** [comment]
Thanks @romainneutron

**Pull request merged**